go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -358,58 +420,74 @@ func (npw *nodePortWatcher) updateServiceInfo(index ktypes.NamespacedName, servi
 		npw.serviceInfo[index].etpHostRules = *etpHostRules
 	}
 
+	if etpNonHostRules != nil {
+		npw.serviceInfo[index].etpNonHostRules = *etpNonHostRules
+	}
+
 	return old, exists
 }
 
 // addServiceRules ensures the correct iptables rules and OpenFlow physical
-// flows are programmed for a given service and hostNetwork endpoint configuration
-func addServiceRules(service *kapi.Service, hasHostNet bool, npw *nodePortWatcher) {
-	if util.ServiceExternalTrafficPolicyLocal(service) && hasHostNet {
-		klog.V(5).Infof("Adding externalTrafficPolicy:local and hostNetworked rules for %v", service)
-		// For dpu or Full mode
-		if npw != nil {
-			npw.updateServiceFlowCache(service, true, true)
-			npw.ofm.requestFlowSync()
-			// Dont touch iptables if in dpuMode
-			if !npw.dpuMode {
-				addSharedGatewayIptRules(service, true)
-			}
-			return
-		}
-		// For Host Only Mode
-		addSharedGatewayIptRules(service, true)
-	} else {
-		// For dpu or Full mode
-		if npw != nil {
-			npw.updateServiceFlowCache(service, true, false)
-			npw.ofm.requestFlowSync()
-			if !npw.dpuMode {
-				addSharedGatewayIptRules(service, false)
-			}
-			return
+// flows are programmed for a given service and endpoint configuration
+func addServiceRules(service *kapi.Service, svcHasLocalHostNetEndPnt, svcHasNonLocalHostNetEndPnt bool, npw *nodePortWatcher) {
+	addFlows := true
+	// For dpu or Full mode
+	if npw != nil {
+		npw.updateServiceFlowCache(service, addFlows, svcHasLocalHostNetEndPnt, svcHasNonLocalHostNetEndPnt)
```

Comment:
svcHasNonLocalHostNetEndPnt reads like has non local, host networked endpoints. I would try to to avoid using negation in the variable name here and say svcHasOVNNetEp

Reply:
> svcHasNonLocalHostNetEndPnt reads like has non local, host networked endpoints. 

that's exactly what its supposed to mean :D